src/controllers/colorjsproxy.cpp

```diff
@@ -1,11 +1,12 @@
 #include "controllers/colorjsproxy.h"
 #include "preferences/hotcuecolorpalettesettings.h"
 
-ColorJSProxy::ColorJSProxy(
-        QScriptEngine* pScriptEngine, UserSettingsPointer pConfig)
+ColorJSProxy::ColorJSProxy(QScriptEngine* pScriptEngine,
+        HotcueColorPaletteSettings colorPaletteSettings)
         : m_pScriptEngine(pScriptEngine),
-          m_hotcueColorPalette(makeHotcueColorPalette(pScriptEngine, pConfig)),
-          m_pConfig(pConfig) {
+          m_hotcueColorPalette(
```

Comment:
The naming of this variable is somewhat confusing. It is not immediately obvious that it is a QJSValue instead of a HotcueColorPalette. How about renaming it to `m_JsHotcueColorPalette`?